static/js/src/advantage/react/components/Subscriptions/SubscriptionDetails/DetailsContent/DetailsContent.tsx

```diff
@@ -4,54 +4,75 @@ import {
   Col,
   ColProps,
   Row,
+  Spinner,
 } from "@canonical/react-components";
-import React from "react";
+import classNames from "classnames";
+import { UserSubscriptionType } from "advantage/api/types";
+import { useUserSubscriptions } from "advantage/react/hooks";
+import { selectFreeSubscription } from "advantage/react/hooks/useUserSubscriptions";
+import { formatDate } from "advantage/react/utils";
+import React, { ReactNode } from "react";
 
 import DetailsTabs from "../DetailsTabs";
 
 type Feature = {
+  valueClassName?: string;
   size?: ColProps["size"];
   title: string;
-  value: string | number;
+  value: ReactNode;
 };
 
 const generateFeatures = (features: Feature[]) =>
-  features.map(({ size = 3, title, value }) => (
+  features.map(({ size = 3, title, value, valueClassName }) => (
     <Col key={title} medium={2} size={size} small={2}>
       <p className="u-text--muted u-no-margin--bottom">{title}</p>
-      <p className="u-no-margin--bottom u-sv1">{value}</p>
+      <p
+        className={classNames("u-no-margin--bottom u-sv1", valueClassName)}
+        data-test={`${title.toLowerCase().replace(" ", "-")}-col`}
+      >
+        {value}
+      </p>
     </Col>
   ));
 
 const DetailsContent = () => {
+  const { data: subscription, isLoading } = useUserSubscriptions({
+    // TODO: Get the selected subscription once the subscription token is available.
+    select: selectFreeSubscription,
+  });
+  const isFreeSubscription = subscription?.type === UserSubscriptionType.Free;
```

Comment:
Here the subscription returned will always be the free one as specified in the select. So I don't see the point of the check right after.
I assume you plan on iterating on every subscription later on and remove the select?

Reply:
Yeah that's right, the token from the selected sub from the subs list will be stored in state. At the moment the API doesn't return the token so there's no way of identifying the subs so I'm just selecting the free one for now.